hngprojects/hng_boilerplate_java_web
# feat: create organisation by users

## How should this be manually tested?
- Try accessing the `/api/v1/organisations` route as an authorized user (Bearer Token).
- Submit a valid request to create an organisation.
- The expected result is to get a `201 Created` response with the organisation details.
- Error responses to check:
  - `403 Unauthorized` if the user is not authenticated.
  - `422 Unprocessable Entity` if there are validation errors.
  -  `400 Bad Request` if the organisation name already exists.

## Checklist of what you did
- [x] Implemented authentication middleware to ensure the user is authenticated using JWT.
- [x] Created the `/api/v1/organisations` endpoint to handle organisation creation.
- [x] Added input validation for all supplied fields in the request body.
- [x] Implemented error handling for various scenarios including name conflicts, validation errors, unauthorized access, and internal server errors.
- [x] Ensured the response structure is consistent for successful and error scenarios.
- [x] Added unit tests and e2e tests for all the features as described more in the reference issue below.
## Reference Issue
[https://github.com/hngprojects/hng_boilerplate_java_web/issues/21]()
